crates/core/src/item.rs

```diff
@@ -27,7 +27,7 @@ const TOP_LEVEL_ATTRIBUTES: [&str; 8] = [
 /// `Item` is the core object in a STAC catalog, containing the core metadata that
 /// enables any client to search or crawl online catalogs of spatial 'assets'
 /// (e.g., satellite imagery, derived data, DEMs).
-#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, SelfHref, Links, Migrate)]
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, SelfHref, Links, Migrate, Fields)]
```

Comment:
This should be on `Properties`, not on `Item`

Reply:
Sorry, got confused since `additional_items` exist on both.

But this makes me think that with the current macro definition, I'd rather leave everything as is and not use derive. See examples of the API change to access fields below:

Without this PR:
```
item.fields()
```

With `#[derive(...,Fields]` on `Properties` trait
```
item.properties.fields()
```

As far as I understand, without rewriting the macro, there is no way to derive it on `Item` struct to support the API we have now. If this is correct, I propose to abandon the PR and leave the code as is. I'd rather have one manual implementation and simple macro than one complicated macro.

Reply:
Ah yeah, you're right! Thanks for pushing back.